### ISSUE TITLE
Fix `at-rule-descriptor-no-unknown` false positives for declarations with feature-value-blocks

### ISSUE THREAD

**Issue by kryops** [stylelint/stylelint#8865]
### What minimal example or steps are needed to reproduce the bug?

CSS from the Inter font definition: https://rsms.me/inter/

```css
@font-feature-values InterVariable {
  @character-variant {
    cv01: 1;
    cv02: 2;
    cv03: 3;
    cv04: 4;
    cv05: 5;
    cv06: 6;
    cv07: 7;
    cv08: 8;
    cv09: 9;
    cv10: 10;
    cv11: 11;
    cv12: 12;
    cv13: 13;
    alt-1: 1; /* Alternate one */
    alt-3: 9; /* Flat-top three */
    open-4: 2; /* Open four */
    open-6: 3; /* Open six */
    open-9: 4; /* Open nine */
    lc-l-with-tail: 5; /* Lower-case L with tail */
    simplified-u: 6; /* Simplified u */
    alt-double-s: 7; /* Alternate German double s */
    uc-i-with-serif: 8; /* Upper-case i with serif */
    uc-g-with-spur: 10; /* Capital G with spur */
    single-story-a: 11; /* Single-story a */
    compact-lc-f: 12; /* Compact f */
    compact-lc-t: 13; /* Compact t */
  }

  @styleset {
    ss01: 1;
    ss02: 2;
    ss03: 3;
    ss04: 4;
    ss05: 5;
    ss06: 6;
    ss07: 7;
    ss08: 8;
    open-digits: 1; /* Open digits */
    disambiguation: 2; /* Disambiguation (with zero) */
    disambiguation-except-zero: 4; /* Disambiguation (no zero) */
    round-quotes-and-commas: 3; /* Round quotes &amp; commas */
    square-punctuation: 7; /* Square punctuation */
    square-quotes: 8; /* Square quotes */
    circled-characters: 5; /* Circled characters */
    squared-characters: 6; /* Squared characters */
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
	extends: ['stylelint-config-standard'],
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAzCA7ALgWhTAhpgK4BOM2AbvgDZEwDOABAJJYwkBq+JAlvgEbUYjYAB10jRkjAALbvjCZ2lbnywjxkyWAoAGAIzxG+gNyatO3QCYjVsxIt6AzEaf2tjSwBYjX9490AViNA-209ADYjCLDPPQB2I3jYywAOI1SUvQBOI2ys-V0jQoLDY1NzcP0bYztKuP0XYzd6mhwy00YAegAqRgBBaiUSdEJhDGEertah7Cb87r6AMWpCbEwIAAdGTBkySemHSS2YdGwfRjtFxgB5TdPGNFJGKfqTs6jGN2u7h-oeAAeL0OHne2FyjD8P3uEnQPHQB3q1DA2Go2AA7jxdut8DxqCETNcADIQdHKMD4ejCImMTG7Ha46jA+r-AC2m2oPBQPBgUGwRGihN6jAAyjx2ZzubzGERmUdGG1sFAIERBOR6EkhX1BsNRkpGABxdis-ASZWqoSMJiveVEFE8DFYmTYKm8FAZLWMACqm3uJGwFKpjB4tKdVvYXLlHjt2AA5o7sfRNqRirpPQBhfCbLE0Q2h+lJ542jz-dCxoQujYkACe2HwxU6wrFZYr9Cr1YVUYsEHZChwyNwxSuwvTPc2fceXe0Y77qJRmGK3xHM8UOy7AF9xOYkG3q0IqZgNPL6PQDMVYifrLYL6emi1j6eLn4WafgoxQi-dJ8Yp-Eoxkp+6SMJkbwwkqPCxliGrGJ6vxmhBUFTowUA8PQ+CsvwEFEIQPAYNe1wACKoehmGxthmC4RIAAUdIyIwABe7AQAAlEhKFoRhWE4Rg2AwACYAwJsOCMSQEC+J6REcaR5GUYwVHoBADFMaxxZaKJRDoHyACORAQEo9B1ppAY9ia0FLn0ABKKqaYwOl6QwjAAGToZshKQKyplIfQOncOQyboIoMl4f+noij5ZCMP5gXcRIqmSN52FkNgdn6R61xhYlwgpQ5cWeDwJBgEIfKyPIijsNBoTXOm+WFdKJUkH25VeeFvIBnIDVlSQ0ExOlLVQJ47WNV1G7iOuIAADQgNyQgAHLoXAiB8S5QgAHRgCeE3gBg3KxggIB8ZsEAkIesAoPgRBDEeohKACSiadBADaADku5CJyWDGegO2VqaUDcFAT0ALrjaN9ibZAX0QUsR0mpge2vTA72YGt20QStrIAFb0JtsCbNjiBiA4oggAjSPE0YxOrPpmDEyDRMk5ge6I-COAQ99ba-f95OMJTYxtsTo0gOuQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.26.1",
  "stylelint-config-css-modules": "4.5.1",
  "stylelint-config-standard": "39.0.1"
}
```

### What did you expect to happen?

No error

### What actually happened?

Errors for every property:

```
Unexpected unknown descriptor "cv01" for at-rule "@font-feature-values" (at-rule-descriptor-no-unknown)Stylelint[at-rule-descriptor-no-unknown](https://stylelint.io/user-guide/rules/at-rule-descriptor-no-unknown)
```

### Do you have a proposal to fix the bug?

16.26.0 did not report an error here; maybe regression in https://github.com/stylelint/stylelint/pull/8850 ?

Thanks for looking into it!

**Comment by jeddy3** [comment]
@kryops Thanks for the clear report and for using the template.

It seems the rule is treating [declarations within feature-value-blocks](https://drafts.csswg.org/css-fonts/#font-feature-values-syntax) as potential descriptors, when it should ignore them.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8868
